crates/autopilot/src/database/order_events.rs

```diff
@@ -18,21 +19,31 @@ impl super::Postgres {
 }
 
 pub async fn store_order_events(
-    db: &super::Postgres,
+    ex: &mut PgConnection,
     order_uids: Vec<domain::OrderUid>,
     label: OrderEventLabel,
     timestamp: DateTime<Utc>,
 ) -> Result<()> {
```

Comment:
We need to keep returning error here because this function is used by different kind of callers.

Reply:
The point of my remark was that there is no reasonable error handling besides logging stuff. That's why we can do all the error handling internally and don't return any result from this function at all.
That way the call site looks clean and the logging happens in a single location instead of at every call site.

Reply:
I removed the `Result` and made it panic for any error.

Reply:
Even if we expect this case to happen extremely rarely panicking is not okay here. This would disrupt everything and not just the surrounding DB operation that this change is partly intended to protect. I was thinking about something along those lines:
```rust
pub async fn store_order_events(
    ex: &mut PgConnection,
    order_uids: Vec<domain::OrderUid>,
    label: OrderEventLabel,
    timestamp: DateTime<Utc>,
) {
    let start = Instant::now();
    let count = order_uids.len();

    let insert = async move {
        let mut ex = ex.begin().await?;

        for uid in order_uids {
            let event = OrderEvent {
                order_uid: ByteArray(uid.0),
                timestamp,
                label,
            };

            order_events::insert_order_event(&mut ex, &event).await?;
        }

        ex.commit().await
    };

    match insert.await {
        Ok(_) => tracing::debug!(?label, count, elapsed = ?start.elapsed(), "stored order events"),
        Err(err) => tracing::warn!(?label, count, ?err, "failed to insert order events"),
    }
}
```